git/attribs.go

```diff
@@ -73,7 +73,7 @@ func GetAttributePaths(workingDir, gitDir string) []AttributePath {
 				fields := strings.Fields(line)
 				pattern := fields[0]
 				if len(reldir) > 0 {
-					pattern = filepath.Join(reldir, pattern)
+					pattern = filepath.ToSlash(filepath.Join(reldir, pattern))
```

Comment:
The necessity of this change was prompted by a consistent Windows test failure I was seeing on a VM. Every time I ran the nested `.gitattributes` test, I saw a failure.

It turns out the problem is that `filepath.Join()` uses `\` as the separator character on Windows, rather than `/`. As a result, a `*.txt` pattern in a folder `subtree` would be parsed as `subtree\*.txt`, and instead of acting as a path separator, the `\` would escape the `*` character, so the pattern would then only match a file with the name `subtree*.txt`.

Reply:
Great fix! One nit is that `filepath.Join` calls `filepath.Clean` (see \[[1](https://github.com/golang/go/blob/master/src/path/filepath/path_unix.go#L45)\]), which can have unintended consequences with certain pattern types (i.e., escape sequences are removed, etc.)

How about we shorten this to:

```go
pattern = fmt.Sprintf("%s/%s", reldir, pattern)
```

or:

```go
pattern = strings.Join([]string{reldir, pattern}, "/")
```

Reply:
@ttaylorr Hey! I believe this code was already being used as-is with the `track` command, so I think that was already removing escape characters from patterns. I did some investigation to confirm this, and `git lfs track "*\*.mov"` gets converted to `*/*.mov` in the `.gitattributes` file, and if you manually change it to `*\*.mov`, that's not matched by `track` on a file with the name `a*.mov`.